cmd/thv-operator/pkg/sources/api_toolhive.go

```diff
@@ -0,0 +1,297 @@
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert to ToolHive format: %w", err)
+	}
+
+	// Calculate hash of the raw data for change detection
+	hash := fmt.Sprintf("%x", sha256.Sum256(data))
+
+	// Create and return fetch result
+	return NewFetchResult(toolhiveRegistry, hash, mcpv1alpha1.RegistryFormatToolHive), nil
+}
+
+// CurrentHash returns the current hash of the API response
+func (h *ToolHiveAPIHandler) CurrentHash(ctx context.Context, mcpRegistry *mcpv1alpha1.MCPRegistry) (string, error) {
+	baseURL := h.getBaseURL(mcpRegistry)
+	apiURL := h.buildServersURL(baseURL)
+
+	// Fetch data from API
+	data, err := h.httpClient.Get(ctx, apiURL)
+	if err != nil {
+		return "", fmt.Errorf("failed to fetch from API: %w", err)
+	}
+
+	// Compute and return hash
+	hash := fmt.Sprintf("%x", sha256.Sum256(data))
+	return hash, nil
+}
+
+// getBaseURL extracts and normalizes the base URL
+func (*ToolHiveAPIHandler) getBaseURL(mcpRegistry *mcpv1alpha1.MCPRegistry) string {
+	baseURL := mcpRegistry.Spec.Source.API.Endpoint
+
+	// Remove trailing slash
+	if len(baseURL) > 0 && baseURL[len(baseURL)-1] == '/' {
+		baseURL = baseURL[:len(baseURL)-1]
+	}
+
+	return baseURL
+}
+
+// buildServersURL constructs the URL for listing servers
+func (*ToolHiveAPIHandler) buildServersURL(baseURL string) string {
+	// ToolHive API path: /v0/servers
+	fullURL := baseURL + "/v0/servers"
+
+	parsedURL, err := url.Parse(fullURL)
+	if err != nil {
+		return fullURL
+	}
+
+	// Add format query parameter
+	q := parsedURL.Query()
+	q.Set("format", "toolhive")
+	parsedURL.RawQuery = q.Encode()
+
+	return parsedURL.String()
+}
+
+// buildServerDetailURL constructs the URL for fetching server details
+func (*ToolHiveAPIHandler) buildServerDetailURL(baseURL, serverName string) string {
+	// Construct URL: /v0/servers/{name}?format=toolhive
+	fullURL := fmt.Sprintf("%s/v0/servers/%s", baseURL, url.PathEscape(serverName))
+	parsedURL, err := url.Parse(fullURL)
+	if err != nil {
+		return fullURL
+	}
+
+	// Add format query parameter
+	q := parsedURL.Query()
+	q.Set("format", "toolhive")
+	parsedURL.RawQuery = q.Encode()
+
+	return parsedURL.String()
+}
+
+// convertToToolhiveRegistry converts API response to ToolHive Registry format
+// by fetching detailed information for each server
+//
+//nolint:unparam // error return kept for future extensibility and consistency with interface
+func (h *ToolHiveAPIHandler) convertToToolhiveRegistry(
+	ctx context.Context,
+	baseURL string,
+	response *ListServersResponse,
+) (*registry.Registry, error) {
+	logger := log.FromContext(ctx)
+
+	toolhiveRegistry := &registry.Registry{
+		Version:       "1.0",
+		LastUpdated:   time.Now().Format(time.RFC3339),
+		Servers:       make(map[string]*registry.ImageMetadata),
+		RemoteServers: make(map[string]*registry.RemoteServerMetadata),
+	}
+
+	// Fetch detailed information for each server
+	for _, serverSummary := range response.Servers {
```

Comment:
I guess that since we won't keep the toolhive format around for more it doesn't make sense to parallelize this? Maybe worth adding a TODO?